### ISSUE TITLE
Nullable Enum won't serialize YaxEnum alias

### ISSUE THREAD

**Issue by prismamed** [YAXLib/YAXLib#159]
**YAXLib version**: 3.0.0

**Framework version**:  net5.0

A nullable enum in a Model class won't be serialized with the corresponding YAXenum alias.
Example Model:

```

public class Test
    {
        public string Text { get; set; }
        public TestEnum TestEnum { get; set; }
        public TestEnum? TestEnumNullable { get; set; }
    }

public enum TestEnum
    {
        [YAXEnum("1")]
        EnumOne,
        [YAXEnum("2")]
        EnumTwo,
        [YAXEnum("3")]
        EnumThree
    }


class Program
    {
        static void Main(string[] args)
        {

var test = new Test()
                       {
                           Text = "Hello World", TestEnum = TestEnum.EnumOne,
                           TestEnumNullable = TestEnum.EnumThree
                       };

            var serializer = new YAXSerializer(typeof(Test));
            var xml = serializer.Serialize(test);

        }
    }
}

```

The resulting xml looks like this:

```
<Test>
  <Text>Hello World</Text>
  <TestEnum>1</TestEnum>
  <TestEnumNullable>EnumThree</TestEnumNullable>
</Test>
```

The expected result would be:

```
<Test>
  <Text>Hello World</Text>
  <TestEnum>1</TestEnum>
  <TestEnumNullable>3</TestEnumNullable>
</Test>
```

I didn't found a workaround.

**Comment by axunonb** [comment]
The wrong value is assigned here

https://github.com/YAXLib/YAXLib/blob/31883783dcc66e84ecf65dfb6fc1c1bef1f6eaed/YAXLib/YAXSerializer.cs#L681

**Comment by axunonb** [comment]
@prismamed Daniel, your feedback on PR #160 would be highly appreciated. Thanks for pointing this out.

**Comment by 304NotModified** [comment]
Fixed in 3.0.1 :)

Great work @axunonb
